wandb/sdk/artifacts/storage_handlers/gcs_handler.py

```diff
@@ -193,3 +203,21 @@ def _entry_from_obj(
             size=obj.size,
             extra={"versionID": obj.generation},
         )
+
+    def _is_dir(
+        self,
+        bucket: str,
+        key: str,
+        size: int,
+    ):
```

Comment:
question/suggestion(optional): do we need to pass in `bucket, key, size` separately?  E.g. seems like there'd be fewer moving parts (so less prone to future bugs/errors) if we 
- only passed in `manifest_entry`, and 
- derived `bucket, key, size` inside `_is_dir()` here

to be clear, this is optional: I realize there's a potential perf impact if  from parsing `bucket, key` extra times, if inside a long loop.   (some kind of caching might fix this, but that seems out of scope for the PR)

Reply:
yeah I think this makes sense, don't think it'll slow down performance in a noticeable way to just derive bucket and key again